src/util/newconfig/ConfigConstraints.hpp

```diff
@@ -0,0 +1,401 @@
+     *
+     * @param type The value type to be checked.
+     * @return true if the type is correct, false otherwise.
+     */
+    [[nodiscard]] bool
+    checkType(ValueType const& type) const override
+    {
+        return std::holds_alternative<int64_t>(type) || std::holds_alternative<std::string>(type);
+    }
+
+    uint32_t const portMin = 1;
+    uint32_t const portMax = 65535;
+};
+
+/**
+ * @brief A constraint to ensure the channel name is valid.
+ */
+class ChannelNameConstraint final : public Constraint {
+public:
+    /**
+     * @brief Check if the channel name is valid.
+     *
+     * @param channel The channel name to check.
+     * @return An optional Error if the channel name is not valid.
+     */
+    [[nodiscard]] std::optional<Error>
+    checkConstraint(ValueType const& channel) const override;
+
+private:
+    /**
+     * @brief Check if the type of the value is correct for this specific constraint.
+     *
+     * @param type The value type to be checked.
+     * @return true if the type is correct, false otherwise.
+     */
+    [[nodiscard]] bool
+    checkType(ValueType const& type) const override
+    {
+        return std::holds_alternative<std::string>(type);
+    }
+};
+
+/**
+ * @brief A constraint to ensure the log level name is valid.
+ */
+class LogLevelNameConstraint final : public Constraint {
+public:
+    /**
+     * @brief Check if the log level name is valid.
+     *
+     * @param logger The log level name to check.
+     * @return An optional Error if the log level name is not valid.
+     */
+    [[nodiscard]] std::optional<Error>
+    checkConstraint(ValueType const& logger) const override;
+
+private:
+    /**
+     * @brief Check if the type of the value is correct for this specific constraint.
+     *
+     * @param type The value type to be checked.
+     * @return true if the type is correct, false otherwise.
+     */
+    [[nodiscard]] bool
+    checkType(ValueType const& type) const override
+    {
+        return std::holds_alternative<std::string>(type);
+    }
+};
+
+/**
+ * @brief A constraint to ensure the IP address is valid.
+ */
+class ValidIPConstraint final : public Constraint {
+public:
+    /**
+     * @brief Check if the IP address is valid.
+     *
+     * @param ip The IP address to check.
+     * @return An optional Error if the IP address is not valid.
+     */
+    [[nodiscard]] std::optional<Error>
+    checkConstraint(ValueType const& ip) const override;
+
+private:
+    /**
+     * @brief Check if the type of the value is correct for this specific constraint.
+     *
+     * @param type The value type to be checked.
+     * @return true if the type is correct, false otherwise.
+     */
+    [[nodiscard]] bool
+    checkType(ValueType const& type) const override
+    {
+        return std::holds_alternative<std::string>(type);
+    }
+};
+
+/**
+ * @brief A constraint to ensure the Cassandra name is valid.
+ */
+class CassandraName final : public Constraint {
```

Comment:
For expansibility, we can use something like DBTypeConstraint to check if the type is one of the candidates. Or use more general constraint like "OneOf" to check the value must be one of the provided values.